test/smt/smt-integration.test.ts

```diff
@@ -109,81 +76,112 @@ describe("State Migration to SMT test", function () {
     expect(r1).to.equal(rootT).to.equal(rootB);
   });
 
-  it("should only writer to be able call add to smt tree", async () => {
+  it("should only writer or owner to be able call add to smt tree", async () => {
     const [owner, addr1, addr2] = await ethers.getSigners();
 
-    const abi = poseidonContract.generateABI(2);
-    const code = poseidonContract.createCode(2);
-    const Poseidon2Elements = new ethers.ContractFactory(abi, code, owner);
-    const poseidon2Elements = await Poseidon2Elements.deploy();
-    await poseidon2Elements.deployed();
-
-    const abi3 = poseidonContract.generateABI(3);
-    const code3 = poseidonContract.createCode(3);
-    const Poseidon3Elements = new ethers.ContractFactory(abi3, code3, owner);
-    const poseidon3Elements = await Poseidon3Elements.deploy();
-    await poseidon3Elements.deployed();
-
-    const Smt = await ethers.getContractFactory("Smt");
-    smt = await upgrades.deployProxy(Smt, [
-      poseidon2Elements.address,
-      poseidon3Elements.address,
+    const { poseidon2Elements, poseidon3Elements } = await deployPoseidons(
+      addr1,
+      false
+    );
+    const smt = await deploySmt(
       addr1.address,
-    ]);
-    await smt.deployed();
+      poseidon2Elements.address,
+      poseidon3Elements.address
+    );
 
-    await expect(smt.connect(addr1).add(1, 2)).not.to.be.reverted;
+    await expect(smt.add(1, 2)).not.to.be.reverted;
     await expect(smt.connect(addr2).add(1, 2)).to.be.revertedWith(
       "caller has no permissions"
     );
-    await expect(smt.add(1, 2)).to.be.revertedWith("caller has no permissions");
+    await expect(smt.connect(addr2).add(1, 3)).to.be.revertedWith(
+      "caller has no permissions"
+    );
+
+    await expect(smt.connect(owner).add(1, 4)).not.to.be.revertedWith(
+      "caller has no permissions"
+    );
   });
 });
 
-describe("State SMT integration tests", function () {
+describe("State SMT integration tests", () => {
```

Comment:
A lot of steps in this tree is something that is repeated in the `SmtStateMigration.run()` method in the `smt-state-migration.ts` script.
Can it be reused?